.ruff.toml

```diff
@@ -12,20 +12,45 @@ select = [
   "E",
   "F",
   "W",
-  #"UP",
-  #"PT"
+  "N",
+  "UP",
+  "BLE",
+  "A",
+  "C4",
+  "INP",
+  "PIE",
+  "T20",
+  "PT",
+  "RET",
+  "TID",
+  "PTH",
+  "PD",
+  "PLC",
+  "PLE",
+  "FLY",
+  "NPY",
+  "PERF",
+  "RUF",
 ]
 extend-ignore = [
   "E712",
   "E721",
-  # pycodestyle (E, W)
-  "E501", # LineTooLong                             # TODO! fix
+   # flake8-pie
+  "PIE808", # Disallow passing 0 as the first argument to range
+  # upgrades
+  "UP038", # Use | in isinstance - not compatible with models and is slower
+  # numpy
+  "NPY002", # TODO: migrate from np.random.rand to np.random.Generator
   # pytest (PT)
   "PT001", # Always use pytest.fixture()
```

Comment:
I dunno, it might have to be a work in progress, PY011 has a shit load to unravel

Reply:
What does PY011 do?

Reply:
PY011 is test Value Raises messages should be shorter and there are quite a few throught the testing base

Reply:
Adding things to ignores is what they are for, just because there's a linter rule for it doesn't make it "correct" there are some rules in ruff which directly conflict with other ruff rules.

These pytest rules are the ones I ignore because they aren't the style we use anywhere in sunpy so why on earth would we adopt it now?

Reply:
So what? It's not the style we want to have?